contracts/LiquidityBridgeContract.sol

```diff
@@ -235,35 +259,66 @@ contract LiquidityBridgeContract {
         uint256 height
     ) public returns (int256) {
         bytes32 quoteHash = hashQuote(quote);
+		
+		// SDL: I don't remember why this should be signed...
         require(verify(quote.liquidityProviderRskAddress, quoteHash, signature), "Invalid signature");
 
+		// SDL: Because a bug in the bridge, only the Low significant 4 bytes
+		// in height are considered. To protect from unintended vulns,
+		// we should require that height < 2^32.
+		
+		// SDL: rename transferredAmount to transferredAmountOrErrorCode
+		// Too many future checks depends on this variable being an error code
+		// Do not hide this fact.
         int256 transferredAmount = registerBridge(quote, btcRawTransaction, partialMerkleTree, height, quoteHash);
 
+		// Return codes should be stored in consts. 
         require(transferredAmount != -303, "Failed to validate BTC transaction");
         require(transferredAmount != -302, "Transaction already processed");
         require(transferredAmount != -304, "Invalid transaction value");
 
+		
         if (shouldPenalize(quote, transferredAmount, callRegistry[quoteHash].timestamp, height)) {
             collateral[quote.liquidityProviderRskAddress] -= quote.penaltyFee;
             emit Penalized(quote.liquidityProviderRskAddress, quote.penaltyFee, quoteHash);
             
             // pay reward to sender
+			// SDL: rewardR is a divisor... why not making more granual, such as 
+			// multiplying penaltyFee by 10 before dividing ?
```

Comment:
that's what I was saying (although not so clearly) , but generally it's better to multiply to prevent underflows (depends on args size not to overflow, of course)

penaltyFee  * rewardR /100